stdio/format.c

```diff
@@ -483,7 +479,20 @@ static void format_sprintf_num(void *ctx, feedfunc feed, uint64_t num64, uint32_
 		}
 	}
 
-	const int digits_cnt = tmp - tmp_buf;
+	if ((flags & FLAG_OCT) != 0 && (num64 != 0 || float_frac_len == 0) && (flags & FLAG_ALTERNATE) != 0) {
```

Comment:
MISRA loves its parentheses
```suggestion
	if ((flags & FLAG_OCT) != 0 && ((num64 != 0) || (float_frac_len == 0)) && ((flags & FLAG_ALTERNATE) != 0)) {
```